sh_scrapy/settings.py

```diff
@@ -76,6 +76,12 @@ def _get_action_on_missing_addons(settings):
     return on_missing_addons
 
 
+def _update_component_order(components, path, order):
+    updated_path = update_classpath(path)
+    if not (path in components or updated_path in components):
```

Comment:
It's not sufficient I think, if there's an outdated path in `*_BASE` settings - you'll not have a match and you'll set a duplicate mw/extension here. E.g. in settings.py (it's not prohibited to override `*_BASE` in project settings, note that I used old path to autothrottle ext with `.contrib.`)

```
EXTENSIONS_BASE = {
    'scrapy.extensions.feedexport.FeedExporter': 0, 
    'scrapy.extensions.memdebug.MemoryDebugger': 0, 
    'scrapy.extensions.memusage.MemoryUsage': 0, 
    'scrapy.extensions.logstats.LogStats': 0, 
    'scrapy.extensions.debug.StackTraceDump': 0, 
    'scrapy.telnet.TelnetConsole': 0, 
    'scrapy.extensions.corestats.CoreStats': 0, 
    'scrapy.extensions.closespider.CloseSpider': 0, 
    'sh_scrapy.extension.HubstorageExtension': 100, 
    'scrapy.extensions.spiderstate.SpiderState': 0, 
    'scrapy.contrib.throttle.AutoThrottle': 0
}
```

result after the settings merge https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/pull/25/files#diff-8291937b18d088dbcac48282fe9f81d2R141 (note there're 2 Autothrottle extensions)

```
(Pdb) settings.getdict('EXTENSIONS_BASE')
{'scrapy.contrib.throttle.AutoThrottle': 0, 'scrapy.extensions.corestats.CoreStats': 0, 'scrapy.extensions.debug.StackTraceDump': 0, 'scrapy.extensions.memusage.MemoryUsage': 0, 'scrapy.telnet.TelnetConsole': 0, 'scrapy.extensions.logstats.LogStats': 0, 'scrapy.extensions.feedexport.FeedExporter': 0, 'scrapy.extensions.memdebug.MemoryDebugger': 0, 'scrapy.extensions.closespider.CloseSpider': 0, 'sh_scrapy.extension.HubstorageExtension': 100, 'scrapy.extensions.spiderstate.SpiderState': 0, u'scrapy.extensions.throttle.AutoThrottle': 0}
```